streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImpl.java

```diff
@@ -341,7 +341,7 @@ private void reprocessState(final List<TopicPartition> topicPartitions,
 
                 offset = getGlobalConsumerOffset(topicPartition);
 
-                stateRestoreListener.onBatchRestored(topicPartition, storeName, offset, restoreCount);
+                stateRestoreListener.onBatchRestored(topicPartition, storeName, offset, records.count());
```

Comment:
I don't think this is the right fix. Note, that we check for `record.key != null` -- so I think we need to add a second counter for the for-loop to only count how many record got effectively restored? Cf `restoreState` method below.